### ISSUE TITLE
redaction support

### ISSUE THREAD

**Issue by ansel1** [unidoc/unidoc#194]
Does the library support the primitives required to implement some kind of redaction function?  I'm figuring it would require:

- producing a stream of text content which contains sufficient information to map text sequences back to their underlying tokens
- a method to calculate a rectangle around a sequence of text (for the purpose of creating a rectangular annotation over the region on the page where the text is)
- Some way of altering the original text it can't be recovered, but is replaced with something that doesn't disturb the layout of the rest of the page.

**Comment by gunnsth** [comment]
@ansel1 Yes this should be possible with our current functionality, as we can search and manipulate the content stream.

Can you clarify a few things,
1. Are you looking to replace certain text, i.e. do a search and replace?  or
2. Are you specifying coordinates, specifying a rectangle where text needs to be removed?

Can you give an example with screenshots of what you would like to achieve? i.e. a before and after image?

I realize that we also need to make sure that the original text is gone (e.g. not just hidden under a white rectangle).
